src/runtime/server/plugin.ts

```diff
@@ -19,24 +19,25 @@ export default defineNitroPlugin((nitroApp) => {
     ...rest,
     serverImages: serverImages.map(r => new RegExp(r)),
   }
-  nitroApp.hooks.hook('afterResponse', async (_event) => {
-    // we should add the Vary header to the response: is there a way to check if the response has been committed?
+  const critical = !!options.critical
+  const device = options.device.length > 0
+  const network = options.network.length > 0
+  const detect = options.detectOS || options.detectBrowser || options.userAgent.length > 0
+
+  // todo: remove this just for testing purposes
+  nitroApp.hooks.hook('afterResponse', async (event) => {
+    // We should add the Vary header to the response: is there a way to check if the response has been committed?.
+    // I guess the vary header whould be added by the consule, there are a lot of header here to handle.
```

Comment:
`// I guess the vary header whould be added by the consumer, there are a lot of header here to handle.`